src/gt4py/next/program_processors/runners/dace_iterator/__init__.py

```diff
@@ -118,12 +118,13 @@ def _ensure_is_on_device(
 
 
 def get_connectivity_args(
-    neighbor_tables: Mapping[str, common.NeighborTable],
+    offset_provider: dict[str, Any],
     device: dace.dtypes.DeviceType,
 ) -> dict[str, Any]:
     return {
-        connectivity_identifier(offset): _ensure_is_on_device(offset_provider.table, device)
-        for offset, offset_provider in neighbor_tables.items()
+        connectivity_identifier(offset): _ensure_is_on_device(table.table, device)
+        for offset, table in offset_provider.items()
+        if isinstance(table, common.NeighborTable)
```

Comment:
OPTIONAL:
I do not like the idea of filtering out some arguments.
I mean what could `offset_provider` by than _the_ offset provider.
Also the function now has some overlap with `filter_neighbor_tables()`.